### PR TITLE
Integrate various fixes and updates to main

### PR DESCRIPTION
**Pull request by acraugh**
Various fixes and checks to bring multiple revision branches into a common set.

## Summary

Typo corrections, documentation updates, regression test updates, and a review of new elements added to the IngestLDD. Note that additional action is needed for the IngestLDD definitions marked "INADEQUATE" before the branch can be merged and the dictionary released.

## Test Data and/or Report

See test/

## Related Issues

Dictionary is in active development - no issues filed.

### CODE REVIEW

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml:23]: Should add `MKC        M. Kate Crombie, Indigo Information Systems, LLC` for completeness

CC @katecrombie

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml:165]: Note: this conflicts with 41e1af85df964a9accebeac2a9009f483f4ef575, which is part of what we were trying to avoid by doing this development collaboratively in close-to-real time... But everyone needs to be on board with that strategy in order for it to work.

We will need to sort such things out.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml:250]: Same comment as above for LORRI regarding 41e1af8

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: "CCD" was already present for MVIC. It seems we added "Frame Transer CCD" for LORRI, maybe having glossed over the fact that there was already a CCD value available.

@katecrombie: are you on board with striking "Frame Transfer CCD" and using "CCD" for everything with a CCD?

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Will do

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: PICNIC isn't an acronym (or at least if it is, I've never seen it spelled out). It's the name of a specific detector/product made by Rockwell. Some info [here](https://people.ast.cam.ac.uk/~optics/tech/hawaii/picnic.htm) and [here](https://web.archive.org/web/19970222122800/http://creation.risc.rockwell.com:80/779/mct_fpa/picnic.html)

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: This is in fact a duration, not technically a rate, so units of time are correct. It is slightly different than exposure time (in principle at least, if not in practice)&mdash;it is defined as the time between readouts, not the time the sensor was exposed for.

I will update this description.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Will add details to definition

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: @katecrombie do you remember why this had a minimum of -1?

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Makes sense. Will change.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [reply src/PDS4_NH_IngestLDD.xml]: This might just be a copy/paste issue originating in `nh:(left|right)_side_median_bias`.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Will fix

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Okay. I can add a "d" to the ends of these attribute names.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Will revise

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: Will fix

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [inline src/PDS4_NH_IngestLDD.xml]: I will try to get definitions for these now, but the fact is that TBDs are sometimes unavoidable as a temporary measure. As a mission, we cannot afford (schedule-wise) to put the development of processing and archive preparation software (which depends on established structures and content of things like LDD attributes and allowable values) on hold until publication-ready definitions are produced. While it would be ideal to have fully fleshed-out documentation in place ahead of time, this is not always (maybe never?) the case and delivery deadlines would never be met if software development had to wait for all such things to be finalized.

Yes, TBDs need to be addressed prior to release. It's fairly simple to Ctrl-F "TBD"/"TBS"/"TBR" once things have settled to make sure all of our ducks are sufficiently in a row for public consumption.

It's completely fine with Kate and I if you want to hold things in a staging/fixup/development branch until ready to go, but we need to keep pushing forward on our side. I will look to see if we have anything on hand to inform a definition for these, but no promises.

Comment (cgobat) [reply src/PDS4_NH_IngestLDD.xml]: I could make something up for now, e.g. "Spinning" &rarr; "Spacecraft is spinning" and "3-Axis" &rarr; "Spacecraft pointing is triaxially stable", but the way I see it, that is potentially worse than a TBD because the intent to revisit it later is less clear and it would be much easier for something like that that we *did* want to change to end up slipping through the cracks.

**Comment by acraugh** [comment]
So the problem is, as you demonstrated in an earlier comment, that while I
might think to search for "TBD", how the heck would I know to also search
for "TBR" and whatever the third one was that I can't even remember long
enough to complain about it?

The definitions don't have to be "publication-ready". They have to be good
enough that when the updates are forgotten the reviewers and users do not
find a dictionary full of "TBD" abbreviations. Partial information is
better than none.

For marking things needing update, I would suggest we settle on a string
that will always be used regardless of context - like, say "[FINISH]" or
"[REPLACE]". That is something that is typed intentionally and we can then
know for certain what ONE string to search on to find those things needing
additional detail. No matter how good the intentions, I find things like
"TBD" in published documentation on a regular basis in the PDS. Schedules
are too tight - the pieces never get picked up. A visually distinctive flag
("TBD" is very effectively camouflaged in a document already stuffed to the
gills with acronyms) might at least be a constant visual annoyance that
serves as a reminder.

So, for example, for something like the spin state values, you might have a
placeholder definition like: "[REPLACE] Spin states and their identifiers
will be defined in the Instrument SIS,...." and then name a section or give
a document number. Then, at least, when it is forgotten a seriously
interested user has half a chance of finding the information they need
somewhere. And yes, it might be the same for each value in a case like this.

As for the conflicting changes, I don't recognize them because other folks
introduced them. What's the most efficient way to solve this? We have a
meeting scheduled next week, but that's a bit far off. As far as deadlines
go, ours is in May - so this needs to be solved relatively quickly. We'd
need to change and rerun templates and validation, plus edit any associated
documentation.

I'm out for an urgent doctor's appointment this morning from about 10am-1pm
EDT. I'll try to pull folks in for a 2pm EDT meeting today - it's our usual
time, just a week early. Let's see who's available.

Thanks!

-Anne.

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml:165]: I don't understand why this is a conflict. This appears to reference something in another development branch, not something included in this PR. This is what the biweekly (perhaps they should be weekly?) meetings are supposed to sort out, along with us reviewing each others PRs. Since that branch only appeared yesterday, there hasn't been a meeting yet in which to discuss it...

Comment (cgobat) [reply src/PDS4_NH_IngestLDD.xml:165]: It's a conflict in the sense that this PR contains LORRI content that is different than what was added in the other new branch. If/when we try to merge both things into the same place, there will be a merge conflict and we will have to pick one or the other.

**Comment by acraugh** [comment]
Yes, but that's a conflict that will exist then, not now.  Is it a reason
to hold up this merge?  More specifically, are you planning to make use of
those particular additions immediately?

-Anne.

**Comment by cgobat** [comment]
No, it's not a reason to hold this up. I was just flagging it as something that will need to be discussed/resolved sooner or later.

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml:165]: Noted, and Ben has gotten his heads-up. The value will remain the same, but aspects of the definitions might be merged. That will be a separate action after this merge, though.

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml]: So it is. It is also specific to Rockwell (that is, a trademark). Any chance we could tie it to "Rockwell Scientific Imaging"? Mainly this is for users generations from now who might want a clue for researching the instrument.

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml]: We can leave a flag here, if you like ("[REPLACE]" or maybe "[CHECK]"?) and move on...

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml]: Another place where a flag (probably "[REPLACE]" for this) can be left.

Comment (katecrombie) [reply src/PDS4_NH_IngestLDD.xml]: How about we use [INCOMPLETE] as our flag for definitions that we know we need, but have incomplete or inadequate items?  I like [CHECK] as the flag if we are unsure of the facts given.  That way we KNOW we need to search these two terms and ajudacate any that are remaining prior to LDD publication.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [reply src/PDS4_NH_IngestLDD.xml]: I am currently waiting to hear back from the Ralph instrument team about what units these `*RATE` keywords are in.

File: src/PDS4_NH_IngestLDD.xml
Comment (cgobat) [reply src/PDS4_NH_IngestLDD.xml]: Okay, so just change it to "Rockwell Scientific Imaging PICNIC HgCdTe array"?

File: src/PDS4_NH_IngestLDD.xml
Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml]: If we're likely to be waiting on this for awhile I'm inclined to replace the flag with [INCOMPLETE] and move on. This is the last thing to be resolved prior to merge, as far as I can see.

Comment (acraugh) [reply src/PDS4_NH_IngestLDD.xml]: In fact, I just did that and also rewrote the "coefficient_*" definitions to include context and removed those flags. That clears everything and we're good to merge as far as I'm concerned. I am going to hold off for an hour or so to compose a document for the src/ directory on conventions and style for the IngestLDD so we can have something reasonably consistent. If it works well we can offer it to the rest of PDS.

**Review by cgobat — APPROVED** [review]
Good to go

**Pull request merged**
